src/otg/dataset/l2g_prediction.py

```diff
@@ -62,9 +62,17 @@ def from_credible_set(
         Returns:
             L2GPrediction: L2G dataset
         """
+        gwas_study_locus = StudyLocus(
```

Comment:
this could be a reasonable StudyLocus method. Most likely we will use it again `study_locus.filterByStudyType(study_index, ["gwas"])`.

It's a refactor so it could go in a different PR